packages/flutter/lib/src/services/system_navigator.dart

```diff
@@ -10,16 +10,25 @@ import 'system_channels.dart';
 class SystemNavigator {
   SystemNavigator._();
 
-  /// Instructs the system navigator to remove this activity from the stack and
-  /// return to the previous activity.
+  /// Removes the topmost Flutter instance, presenting what was before
+  /// it.
   ///
-  /// On iOS, calls to this method are ignored because Apple's human interface
-  /// guidelines state that applications should not exit themselves.
+  /// On Android, removes this activity from the stack and returns to
+  /// the previous activity.
   ///
-  /// This method should be preferred over calling `dart:io`'s [exit] method, as
-  /// the latter may cause the underlying platform to act as if the application
-  /// had crashed.
-  static Future<void> pop() async {
-    await SystemChannels.platform.invokeMethod<void>('SystemNavigator.pop');
+  /// On iOS, calls `popViewControllerAnimated:` if the root view
```

Comment:
Can we document this in terms of what happens from a user perspective instead of what APIs are called? Just like its done on Android above.

Reply:
Also: I thought apple doesn't allow popping the root ViewController (see old documentation). What has changed?

Reply:
> Can we document this in terms of what happens from a user perspective instead of what APIs are called? Just like its done on Android above.

The reason I talked about the actual implementation is because "animated" doesn't have a definite meaning.  It means whatever Apple decides it means, and could potentially change it.  You might even be able to customize it.  So, if we said "animated means you will do a slide out animation to the left" that might not be true always.  I could say something like "performs UIViewController's dismiss animation", you like that?

> Also: I thought apple doesn't allow popping the root ViewController (see old documentation). What has changed?

Yep, that changed.  See: https://github.com/flutter/engine/blob/c12b917e95074aec4700dd5554563bc84f93ecc2/shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm#L197

It now has slightly different semantics for iOS since the platforms' concepts don't really match.

Reply:
My comment wasn't talking about the documentation about "animated". That one is fine. It was talking about the "popViewControllerAnimated". That one should say in prose what calling "pop" means on iOS, e.g. the FlutterViewController is removed or whatever it does. Similarly to how the Android part documents this.

The link you provided basically has a comment that describes what happens in words instead of in terms of API.

Reply:
Sorry, github isn't good at displaying chains of PR's.  That text was already approved in a previous PR, that's why I as confused.

> That one should say in prose what calling "pop" means on iOS

I believe that it is already explained on the first line in prose, "Removes the topmost Flutter instance, presenting what was before it.".  

I've added the iOS section about implementation details in order to provide more color.  The comment in the code I link referred to "navigation controllers" which is a less specific way to say `UINavigationController`.  I think in this case when we are providing further details; we are doing users a disservice by avoiding concrete classes and methods.  I'm happy to chat in person if you want to look into it more =)